ext/standard/var_unserializer.re

```diff
@@ -1193,6 +1193,96 @@ object ":" uiv ":" ["]	{
 	return object_common(UNSERIALIZE_PASSTHRU, elements, has_unserialize);
 }
 
+"E:" uiv ":" ["]	{
+	size_t len, maxlen;
+	char *str;
+	zend_string *full_name, *enum_name, *case_name;
+	zend_class_entry *ce;
+
+    if (!var_hash) return 0;
+
+	len = parse_uiv(start + 2);
+	maxlen = max - YYCURSOR;
+	if (maxlen < len || len == 0) {
+		*p = start + 2;
+		return 0;
+	}
+
+	str = (char*)YYCURSOR;
+
+	YYCURSOR += len;
+
+	if (*(YYCURSOR) != '"') {
+		*p = YYCURSOR;
+		return 0;
+	}
+	if (*(YYCURSOR+1) != ';') {
+		*p = YYCURSOR+1;
+		return 0;
+	}
+
+	full_name = zend_string_init(str, len, 0);
+	int32_t colon_pos = -1;
+	for (uint32_t i = 0; i < ZSTR_LEN(full_name); ++i) {
+		if (ZSTR_VAL(full_name)[i] == ':') {
+			colon_pos = i;
+			break;
+		}
+	}
+	ZEND_ASSERT(colon_pos != -1);
+	ZEND_ASSERT(ZSTR_VAL(full_name)[colon_pos + 1] == ':');
+
+	enum_name = zend_string_init_fast(ZSTR_VAL(full_name), -(colon_pos - len) - 2);
+	case_name = zend_string_init_fast(&ZSTR_VAL(full_name)[colon_pos + 2], len - colon_pos - 2);
+	zend_string_release_ex(full_name, 0);
+
+	//if (!zend_is_valid_class_name(enum_name)) {
+	//	zend_string_release_ex(enum_name, 0);
+	//	zend_string_release_ex(case_name, 0);
+	//	return 0;
+	//}
+
+	ce = zend_lookup_class(enum_name);
+	if (!ce) {
+		php_error_docref(NULL, E_WARNING, "Class '%s' not found", ZSTR_VAL(enum_name));
```

Comment:
What's the reason for emitting E_WARNING instead of throwing an Error - there's no bc break for new functionality?

Reply:
`unserialize` doesn't seem to throw ATM at all, we may want to stay consistent here.

Reply:
Oh, right, I forgot that that was one place that didn't become stricter